frontend/public/components/factory/table.tsx

```diff
@@ -328,16 +328,17 @@ const VirtualBody: React.SFC<VirtualBodyProps> = (props) => {
   );
 };
 
-export type RowFunctionArgs = {
-  obj: object;
+export type RowFunctionArgs<T = any, C = any> = {
+  obj: T;
   index: number;
   columns: [];
   isScrolling: boolean;
   key: string;
   style: object;
-  customData?: any;
+  customData?: C;
 };
-export type RowFunction = (args: RowFunctionArgs) => JSX.Element;
+
+export type RowFunction<T = any, C = any> = (args: RowFunctionArgs<T, C>) => JSX.Element;
```

Comment:
@christianvogt Should we change this to `=> React.ComponentType<any>` ?

Reply:
It should return `React.ReactNode`

Reply:
Agreed, let's change this as part of follow-up PR (part 2) which will address the convoluted cases.